client/my-sites/checkout/composite-checkout/components/wp-order-review-line-items.tsx

```diff
@@ -201,6 +202,21 @@ function LineItemWrapper( {
 		if ( isJetpack || isAkismet ) {
 			return true;
 		}
+
+		try {
+			const { variationName } = dangerouslyGetExperimentAssignment( 'calypso_plans_2yr_toggle' );
```

Comment:
Are we sure that we are pre-loading the experiment with enough time for this point? Do we have an estimate?

Reply:
Good question. [The doc](https://github.com/Automattic/wp-calypso/blob/trunk/packages/explat-client/README.md#L50) says:

> minimum 2 seconds looking at perf data, 5-10 seconds is best

If we include the preload in the previous comment, for the signup case we'd load the experiment from `/start` so I think it should be safe anecdotally. However, for the alread logged-in case, we'd load the experiment only from `/plans`, which means they do have a chance to access `/plans` and click on any upgrade button to get to the checkout page sooner. One potential solution is to block loading the pricing grid so it's not possible, even though that's something I intentionally avoid here. 

Could you advice how we could estimate the time?

Reply:
> Could you advice how we could estimate the time?

I think the easiest way is to track timing using developer tools as you run through the experience.
You can see when the API call to `/experiments` occurs, and you can see what time you hit this code with a `console.log`.

> One potential solution is to block loading the pricing grid so it's not possible, even though that's something I intentionally avoid here.

Yeah it's a hard decision to make, but I'd recommend it if the timing is too tight, and there is no other way around that.

Reply:
>  for the signup case we'd load the experiment from /start so I think it should be safe anecdotally

Yes, the assignment is definitely done by the time a `/start` user reaches checkout from loading the signup controller as there are at least 2 steps in between.

> they do have a chance to access /plans and click on any upgrade button to get to the checkout page sooner.

That chance is pretty slim as I see we're actually [displaying a spinner while the assignment is loading](https://github.com/Automattic/wp-calypso/pull/75063/files#diff-55518f6f3437bf6669939fefd8df0377540924ae6a53133830805b30d028f04cR19-R21). It's true this is only for the `<PlanTypeSelector />` but still, it gives the impression that the grid is loading.

@jessie-ross , I wonder if a solution to be 100% sure that all analyzed users were correctly exposed, without adding extra loading time, is to fire a Tracks event when rendering the `<PlanTypeSelector />` so it will say something like `calypso_plans_plan_type_selector_view` and use that as exposure event.

Reply:
> I wonder if a solution to be 100% sure that all analyzed users were correctly exposed, without adding extra loading time, is to fire a Tracks event when rendering the <PlanTypeSelector /> so it will say something like calypso_plans_plan_type_selector_view and use that as exposure event.

Yeah, an exposure event we know only occurs once the experiment has loaded should be unbiased and help out - it would be useful to have in the event things aren't working.

Reply:
@southp, is there a Tracks event that already fires when rendering the `<PlanTypeSelector />`? If not, it sounds like one should be added to be used as an exposure event in this experiment. Let me know what the exposure event ends up being, and I can add it to our experiment!

Reply:
@jessie-ross https://github.com/Automattic/wp-calypso/commit/6e99cf40b4c1eb23340fb8dbb1b78069311ec133 should be following the 3rd option:

> If blocking loading is too complex, try adding an exposure event that fires only if users were exposed (hit the experience after loading), although note that this is complex too.

which is a direct implementation of @razvanpapadopol 's [idea](https://github.com/Automattic/wp-calypso/pull/75063#discussion_r1157855919). Rethinking about it, maybe what's needed here is to add an event that fires when `<TermExperimentPlanTypeSelector/>` correctly loads the experiment and renders the underlying `<PlanTypeSelector/>`, rather than when `<PlanTypeSelector/>` is rendered? Could you help me confirm?

Reply:
@southp not sure if exposure events should depend on experiment events such as the successful loading of an experiment. An exposure event generally symbolises a meaningful app event in the flow of the application. This is important to understand the result of the experiment and maintaining clarity of what exact events were filtered out.

WDYT @jessie-ross @razvanpapadopol ?

Reply:
Hi, @southp and thank you for implementing the new exposure event! 

> maybe what's needed here is to add an event that fires when TermExperimentPlanTypeSelector correctly loads the experiment

Same as @jdc91, I also think the current implementation is correct and it also doesn't need to be cleaned up after the experiment. Other experiments or Tracks funnels can use the event in the future if needed.

As a bonus, now we'll have 1 exposure event instead of 2 in the experiment model in Abacus 😁 

> An exposure event generally is based on a meaningful app event in the flow of a given user journey. This distinction is important to understand the result of the experiment and maintaining clarity of what exact events were filtered out.

Great summary, @jdc91 ! In this case, viewing the PlanTypeSelector is the app event that we want to be considered the start of the forked experience. 

As a side note, we are also using the exposure event to "reset the clock" of the attribution window so every time the user will see the PlanTypeSelector, a new 72 hours timeframe for a conversion to be recorded ( PCYsg-FqB-p2 )